src/WebJobs.Script.WebHost/Diagnostics/MetricsEventManager.cs

```diff
@@ -26,8 +27,9 @@ public class MetricsEventManager : IDisposable
         private static string appName;
         private static string subscriptionId;
         private bool _disposed;
+        private IMetricsPublisher _metricsPublisher;
 
-        public MetricsEventManager(IEnvironment environment, IEventGenerator generator, int functionActivityFlushIntervalSeconds, int metricsFlushIntervalMS = DefaultFlushIntervalMS)
+        public MetricsEventManager(IEnvironment environment, IEventGenerator generator, int functionActivityFlushIntervalSeconds, IMetricsPublisher metricsPublisher = null, int metricsFlushIntervalMS = DefaultFlushIntervalMS)
```

Comment:
I think you're allowing null just so the test doesn't have to pass in a mock? Let's not take null and just pass in a mock in the tests, so it's clear that this param is required, just like generator, etc.